### DISCUSSION TITLE
How to force a Window to be always in front of other widgets

### DISCUSSION THREAD

**Discussion by felipetesc** [emilk/egui#2140] [Q&A]
Right now, I'm creating a small prototype app using the canvas demo available at egui.rs as a starting point. I'm using a window to show the  main menu and not a panel like is commom, so to say. The problem is happening when I start to draw something inside the canvas , once I click on the canvas area and it get the focus the window with the main menu disapears. Inside the drawing method  I'm drawing the window after the canvas. My question is how can I force the window to be always on top of the canvas ? 

https://user-images.githubusercontent.com/2336812/195209647-7e724e55-b4f1-4c06-84b2-299dde34c18f.mp4

**Comment by smallB007** [comment]
Just a suggestion. I know in egui order of drawing widgets matter. Maybe make sure that the window with menu is drawn after canvas?

**Reply by felipetesc** [reply]
Thank you for the suggestion. I am already  doing that. The previous video was created with two windows. The first window  being drawed contains a menu bar and the second window has the canvas element. The problem happens because once the window with the canvas is clicked and  it acquire the focus it shadows the next one. That could be changed if we had a way to control what order the windows should always have.
